src/core/mount.c

```diff
@@ -55,7 +55,7 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(struct libmnt_iter*, mnt_free_iter);
 static const UnitActiveState state_translation_table[_MOUNT_STATE_MAX] = {
         [MOUNT_DEAD] = UNIT_INACTIVE,
         [MOUNT_MOUNTING] = UNIT_ACTIVATING,
-        [MOUNT_MOUNTING_DONE] = UNIT_ACTIVE,
+        [MOUNT_MOUNTING_DONE] = UNIT_ACTIVATING,
```

Comment:
hmm, not sure i feel about this one. 

So my original intention with all this was that the difference in state handling between mounts made by us and those made by others should be minimal. Or to say this differently: the outside view of whether a mount is established or not is unaffected by the mount binary still running or not: a program that accesses the relevant dir will already see it and that counts. And hence we should consider it running too when processing stuff for others...

But I do agree with your point regarding the private mount options in newer util-linux. As those are not atomically applied we have to wait for mount exiting before proceeding. This change is a gamechanger for us I guess on this.

So yeah, I guess this should go in.